libevmasm/KnownState.cpp

```diff
@@ -376,31 +379,31 @@ KnownState::Id KnownState::applyKeccak256(
 	// unknown or too large length
 	if (!l || *l > 128)
 		return m_expressionClasses->find(keccak256Item, {_start, _length}, true, m_sequenceNumber);
-
+	unsigned length = unsigned(*l);
```

Comment:
Could this lead to a subtle truncation issue? I don't think so because the contract would run out of gas but still something I could think of

```
a := keccak256(0, add(UINTMAX, 1))
```

Reply:
The `if` statement returns if `l` is more than 128.